src/bin/build_octree.rs

```diff
@@ -372,8 +391,9 @@ fn main() {
 
         pool.scoped(|scope| {
             for node in &subsample_nodes {
+                let nodes_sender_clone = nodes_sender.clone();
                 scope.execute(move || {
```

Comment:
This will queue all nodes metadata in memory, since you do not have started a thread yet that consumes them. I think that is fine for now, since we are assuming that we hold them in memory anyways.